### PR TITLE
new end point for "/study/by/{username}/export.pgn"

### PR DESCRIPTION
**Pull request by komalsai5135**

### CODE REVIEW

**Comment by komalsai5135** [comment]
can anyone help me for that one failed test .

**Comment by kraktus** [comment]
Hello, run `make format` and commit the changes

File: berserk/clients/studies.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 from __future__ import annotations
 
-from typing import Iterator
+from typing import Any, Dict, Generator, Iterator
```
Comment (icp1994) [inline berserk/clients/studies.py:3]:
```suggestion
from typing import Any, Dict, Iterator
```

File: berserk/clients/studies.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
-from ..formats import PGN
+from ..formats import PGN, JSON
```
Comment (icp1994) [inline berserk/clients/studies.py:5]:
```suggestion
from ..formats import PGN, NDJSON
```

File: berserk/clients/studies.py [hunk 3]
```diff
@@ -24,3 +24,10 @@ def export(self, study_id: str) -> Iterator[str]:
         """
         path = f"/api/study/{study_id}.pgn"
         yield from self._r.get(path, fmt=PGN, stream=True)
+
+    def get_study_by_username(
```
Comment (icp1994) [inline berserk/clients/studies.py:28]: `client.studies.get_study_by_username` is probably too verbose, I'd rather go with just `by_user`

File: berserk/clients/studies.py [hunk 4]
```diff
@@ -24,3 +24,10 @@ def export(self, study_id: str) -> Iterator[str]:
...
+        self, username: str
+    ) -> Generator[Dict[str, Any], None, None]:
+        # end point for api/study/by/{username}
```
Comment (icp1994) [inline berserk/clients/studies.py:31]: This is not documenting what it does. See other endpoint definitions for example.

File: berserk/clients/studies.py [hunk 5]
```diff
@@ -24,3 +24,10 @@ def export(self, study_id: str) -> Iterator[str]:
...
+        path = f"/api/study/by/{username}"
+        yield from self._r.get(path, fmt=JSON, stream=True)
```
Comment (icp1994) [inline berserk/clients/studies.py:33]:
```suggestion
        yield from self._r.get(path, fmt=NDJSON, stream=True)
```
streaming endpoints are always NDJSON or PGN

File: README.rst [hunk 6]
```diff
@@ -145,6 +145,7 @@ Most of the API is available:
 
     client.studies.export_chapter
     client.studies.export
+    client.studies.get_study_by_username
```
Comment (icp1994) [inline README.rst:148]: see name note below

**Review by icp1994 — CHANGES_REQUESTED** [review]

File: berserk/clients/studies.py [hunk 7]
```diff
@@ -24,3 +24,10 @@ def export(self, study_id: str) -> Iterator[str]:
         """
         path = f"/api/study/{study_id}.pgn"
         yield from self._r.get(path, fmt=PGN, stream=True)
+
+    def get_study_by_username(
+        self, username: str
+    ) -> Generator[Dict[str, Any], None, None]:
```
Comment (icp1994) [inline berserk/clients/studies.py:30]:
```suggestion
    ) -> Iterator[Dict[str, Any]]:
```

**Comment by kraktus** [comment]
Thanks for the PR, I've made some changes (see commits above)

**Pull request merged**